src/sql/src/plan/query.rs

```diff
@@ -2094,6 +2107,119 @@ fn plan_table_factor(
     )
 }
 
+// The following SQL is equivalent:
+//
+// SELECT
+//     gs1.generate_series, expand.x, expand.n, gs2.generate_series
+// FROM
+//     (SELECT row_number() OVER (), * FROM ROWS FROM (generate_series(1, 2))) AS gs1
+//     FULL JOIN (SELECT row_number() OVER (), * FROM ROWS FROM (information_schema._pg_expandarray(ARRAY[9])))
+//             AS expand ON gs1.row_number = expand.row_number
+//     FULL JOIN (SELECT row_number() OVER (), * FROM ROWS FROM (generate_series(3, 6))) AS gs2 ON
+//             gs1.row_number = gs2.row_number;
+//
+// and:
+//
+// SELECT
+//     *
+// FROM
+//     ROWS FROM (
+//         generate_series(1, 2),
+//         information_schema._pg_expandarray(ARRAY[9]),
+//         generate_series(3, 6)
+//     );
+//
+// This function creates a HirRelationExpr that is identical to what is
+// produced by the former.
+fn plan_rows_from(
+    ecx: &ExprContext,
+    functions: &[TableFunction<Aug>],
+    alias: Option<&TableAlias>,
+) -> Result<(HirRelationExpr, Scope), anyhow::Error> {
+    let mut funcs = Vec::with_capacity(functions.len());
+    let mut project_outputs = Vec::new();
+    let mut last = 0;
+    let name = normalize::unresolved_object_name(functions[0].name.clone())?;
+    for func in functions {
+        let (expr, scope) = plan_table_function(ecx, &func.name, func.alias.as_ref(), &func.args)?;
+        let mut outputs = Vec::with_capacity(scope.items.len());
+        // Re-jigger the outputs so the row number is first.
```

Comment:
Just noting that what "row number" is here is unknown to readers. I think adding some comments up top can make that much clearer at this point, though.